frontend/src/dom-utils.js

```diff
@@ -0,0 +1,34 @@
+/*!
+ * sri sri guru gaurangau jayatah
+ */
+
+const hide = (element) => {
+  element.style.display = 'none';
+};
+
+const show = (element, display = 'block') => {
+  element.style.display = display;
+};
+
+export const dom = {
+  loading: document.getElementById('loading'),
+  searchError: document.getElementById('search-error'),
+  searchErrorMessage: document.getElementById('search-error-message'),
+  filterPanel: document.getElementById('filter-panel'),
+  stats: document.getElementById('stats'),
+  hits: document.getElementById('hits'),
+  pagination: document.getElementById('pagination'),
+  underProgress: document.getElementById('under-progress'),
+  hide,
+  show,
+};
+
+export const restoreSearchWidgets = () => {
+  show(dom.loading);
+  show(dom.filterPanel);
+  hide(dom.stats);
+  show(dom.hits);
+  show(dom.underProgress);
+  show(dom.pagination);
+  hide(dom.searchError);
+};
```

Comment:
Why would we need to manipulate visibility of these elements manually? What was wrong with the default InstantSearch behaviour and, especially, why is it needed for the current task?

Reply:
i am hoping that you can answer to this one in my previous comments.